// @stacks/blockchain-api-client/lib/generated/models/RosettaPublicKey.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Stacks Blockchain API
 * Welcome to the API reference overview for the <a href=\"https://docs.hiro.so/get-started/stacks-blockchain-api\">Stacks Blockchain API</a>.  <a href=\"https://hirosystems.github.io/stacks-blockchain-api/collection.json\" download=\"stacks-api-collection.json\">Download Postman collection</a>
 *
 * The version of the OpenAPI document: STACKS_API_VERSION
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RosettaPublicKeyToJSON = exports.RosettaPublicKeyFromJSONTyped = exports.RosettaPublicKeyFromJSON = exports.RosettaPublicKeyCurveTypeEnum = void 0;
/**
* @export
* @enum {string}
*/
var RosettaPublicKeyCurveTypeEnum;
(function (RosettaPublicKeyCurveTypeEnum) {
    RosettaPublicKeyCurveTypeEnum["secp256k1"] = "secp256k1";
    RosettaPublicKeyCurveTypeEnum["edwards25519"] = "edwards25519";
})(RosettaPublicKeyCurveTypeEnum = exports.RosettaPublicKeyCurveTypeEnum || (exports.RosettaPublicKeyCurveTypeEnum = {}));
function RosettaPublicKeyFromJSON(json) {
    return RosettaPublicKeyFromJSONTyped(json, false);
}
exports.RosettaPublicKeyFromJSON = RosettaPublicKeyFromJSON;
function RosettaPublicKeyFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'hex_bytes': json['hex_bytes'],
        'curve_type': json['curve_type'],
    };
}
exports.RosettaPublicKeyFromJSONTyped = RosettaPublicKeyFromJSONTyped;
function RosettaPublicKeyToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'hex_bytes': value.hex_bytes,
        'curve_type': value.curve_type,
    };
}
exports.RosettaPublicKeyToJSON = RosettaPublicKeyToJSON;
