// @stacks/blockchain-api-client/lib/generated/models/RosettaSignature.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Stacks Blockchain API
 * Welcome to the API reference overview for the <a href=\"https://docs.hiro.so/get-started/stacks-blockchain-api\">Stacks Blockchain API</a>.  <a href=\"https://hirosystems.github.io/stacks-blockchain-api/collection.json\" download=\"stacks-api-collection.json\">Download Postman collection</a>
 *
 * The version of the OpenAPI document: STACKS_API_VERSION
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RosettaSignatureToJSON = exports.RosettaSignatureFromJSONTyped = exports.RosettaSignatureFromJSON = exports.RosettaSignatureSignatureTypeEnum = void 0;
const _1 = require("./");
/**
* @export
* @enum {string}
*/
var RosettaSignatureSignatureTypeEnum;
(function (RosettaSignatureSignatureTypeEnum) {
    RosettaSignatureSignatureTypeEnum["ecdsa"] = "ecdsa";
    RosettaSignatureSignatureTypeEnum["ecdsa_recovery"] = "ecdsa_recovery";
    RosettaSignatureSignatureTypeEnum["ed25519"] = "ed25519";
    RosettaSignatureSignatureTypeEnum["schnorr_1"] = "schnorr_1";
    RosettaSignatureSignatureTypeEnum["schnorr_poseidon"] = "schnorr_poseidon";
})(RosettaSignatureSignatureTypeEnum = exports.RosettaSignatureSignatureTypeEnum || (exports.RosettaSignatureSignatureTypeEnum = {}));
function RosettaSignatureFromJSON(json) {
    return RosettaSignatureFromJSONTyped(json, false);
}
exports.RosettaSignatureFromJSON = RosettaSignatureFromJSON;
function RosettaSignatureFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'signing_payload': (0, _1.SigningPayloadFromJSON)(json['signing_payload']),
        'public_key': (0, _1.RosettaPublicKeyFromJSON)(json['public_key']),
        'signature_type': json['signature_type'],
        'hex_bytes': json['hex_bytes'],
    };
}
exports.RosettaSignatureFromJSONTyped = RosettaSignatureFromJSONTyped;
function RosettaSignatureToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'signing_payload': (0, _1.SigningPayloadToJSON)(value.signing_payload),
        'public_key': (0, _1.RosettaPublicKeyToJSON)(value.public_key),
        'signature_type': value.signature_type,
        'hex_bytes': value.hex_bytes,
    };
}
exports.RosettaSignatureToJSON = RosettaSignatureToJSON;
